File: 프로그래머스/3/43238. 입국심사/입국심사.py
def solution(n, times):
    start, end = 1, max(times) * n
    answer = end
    
    while start <= end:
        mid = (start + end) // 2
        people = sum(mid // time for time in times)

        if people >= n:
            answer = mid
            end = mid - 1
        else:
            start = mid + 1
    
    return answer

File: 프로그래머스/3/43238. 입국심사/입국심사.js
function solution(n, times) {
    let start = 1
    let end = Math.max(...times) * n
    let answer = end;
    
    while (start <= end) {
        let mid = Math.floor((start + end) / 2)
        let total = 0
        // 각 심사관마다 심사할 수 있는 인원
        for (let time of times) {
            total += Math.floor(mid / time)
        }
        
        if (total >= n) {
            answer = mid
            end = mid - 1
        } else {
            start = mid + 1
        }
    }
    
    return answer
}